go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -1067,12 +1201,59 @@ func findLinkOnSameNetworkAsIPUsingLPM(ip net.IP, v4, v6 bool) (bool, netlink.Li
 	return true, link, nil
 }
 
-func getNetlinkAddressWithLabel(addr *net.IPNet, ifindex int, linkName string) *netlink.Addr {
+// getFilteredPrefixes returns address Prefixes from interfaces with the following characteristics:
+// Link must be up
+// Exclude keepalived assigned addresses
+// Exclude addresses assigned by metal LB
```

Comment:
We also exclude link local addresses, may want to update comment here to reflect that